lib/wit/gitrepo.py

```diff
@@ -89,6 +95,17 @@ def clone(self, source, name):
             else:
                 raise
 
+    # Only newer git versions can use '--reference-if-able', so we emulate the 'if-able'.
+    def git_reference_capable(self):
```

Comment:
I think we should 1) give this method a name with a leading underscore to indicate that it is "private" and 2) we should probably give it a name that gives a better hint at what its return value is. "git reference capable" to me sounds like an adjective, and I would expect this to return a boolean value, but you're actually returning a list of command line arguments to append to a `clone` operation.

Since this is a non-mutating method with no arguments and that returns a non-boolean value, I'd probably name it something like `def _git_reference_options()`